projects/hello_buttons_adcs/src/main.c

```diff
@@ -0,0 +1,58 @@
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+GpioAddress adc_pin_addr = { .port = GPIO_PORT_A, .pin = 6 };
+GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings s_adc_pin_settings = {
+  .direction = GPIO_DIR_IN,
+  .state = GPIO_STATE_LOW,
+  .resistor = GPIO_RES_NONE,
+  .alt_function = GPIO_ALTFN_ANALOG,
+};
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,       //
+  .priority = INTERRUPT_PRIORITY_NORMAL,  //
+};
+
+static AdcChannel s_adc_pin_channel = NUM_ADC_CHANNELS;
+
+static void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  uint16_t adc_pin_data = 0;
+
+  adc_read_converted(s_adc_pin_channel, &adc_pin_data);
+  LOG_DEBUG("ADC data: %d\n", adc_pin_data);
+}
+
+static void prv_register_interrupts(void) {
+  gpio_it_register_interrupt(&button_addr, &s_interrupt_settings, INTERRUPT_EDGE_FALLING,
+                             prv_button_interrupt_handler, (void *)0);
```

Comment:
Please use `NULL` instead of `(void *)0`. (Although I did prefer your solution of passing around a pointer to `s_adc_pin_channel` as the context, generally it's a good idea to minimize the use of global variables as much as possible.)